--- a/bin/python/Makefile.in
+++ b/bin/python/Makefile.in
@@ -53,9 +53,9 @@ install:: ${TARGETS} installdirs
 	${INSTALL_DATA} ${srcdir}/dnssec-keymgr.8 ${DESTDIR}${mandir}/man8
 	if test -n "${PYTHON}" ; then \
 		if test -n "${DESTDIR}" ; then \
-			${PYTHON} ${srcdir}/setup.py install --root=${DESTDIR} --prefix=${prefix} ; \
+			${PYTHON} ${srcdir}/setup.py install --root=${DESTDIR} --prefix=${prefix} --install-layout=deb ; \
 		else \
-			${PYTHON} ${srcdir}/setup.py install --prefix=${prefix} ; \
+			${PYTHON} ${srcdir}/setup.py install --prefix=${prefix} --install-layout=deb ; \
 		fi \
 	fi
 
