diff -ur lib/dns/Makefile lib/dns/Makefile
--- lib/dns/Makefile	2016-03-16 19:44:06.100863999 +0000
+++ lib/dns/Makefile	2016-03-16 20:03:45.980863999 +0000
@@ -113,9 +113,9 @@
 
 CWARNINGS =
 
-ISCLIBS =	../../lib/isc/libisc.la
+ISCLIBS =	../../lib/isc/libisc-export.la
 
-ISCDEPLIBS =	../../lib/isc/libisc.la
+ISCDEPLIBS =	../../lib/isc/libisc-export.la
 
 LIBS =		-ldl  -L../../lib/isc -lcrypto
 
diff -ur lib/irs/Makefile lib/irs/Makefile
--- lib/irs/Makefile	2016-03-16 19:44:06.104863999 +0000
+++ lib/irs/Makefile	2016-03-16 20:04:10.700863999 +0000
@@ -101,7 +101,7 @@
 		gai_strerror.c getaddrinfo.c getnameinfo.c \
 		resconf.c
 
-LIBS =		-ldl  -L../../lib/isc -L../../lib/dns -L../../lib/isccfg -lcrypto -lisc -ldns -lisccfg
+LIBS =		-ldl  -L../../lib/isc -L../../lib/dns -L../../lib/isccfg -lcrypto -lisc-export -ldns-export -lisccfg-export
 
 SUBDIRS =	include
 TARGETS =	timestamp
diff -ur lib/isccc/Makefile lib/isccc/Makefile
--- lib/isccc/Makefile	2016-03-16 19:44:06.108863999 +0000
+++ lib/isccc/Makefile	2016-03-16 20:04:22.212863999 +0000
@@ -89,10 +89,10 @@
 CDEFINES =
 CWARNINGS =
 
-ISCLIBS =	../../lib/isc/libisc.la
+ISCLIBS =	../../lib/isc/libisc-export.la
 ISCCCLIBS =	../../lib/isccc/libisccc-export.la
 
-ISCDEPLIBS =	../../lib/isc/libisc.la
+ISCDEPLIBS =	../../lib/isc/libisc-export.la
 ISCCCDEPLIBS =	libisccc-export.la
 
 LIBS =		-ldl  -L../../lib/isc
diff -ur lib/isccfg/Makefile lib/isccfg/Makefile
--- lib/isccfg/Makefile	2016-03-16 19:44:06.112863999 +0000
+++ lib/isccfg/Makefile	2016-03-16 20:04:45.976863999 +0000
@@ -89,12 +89,12 @@
 CDEFINES =
 CWARNINGS =
 
-ISCLIBS =	../../lib/isc/libisc.la
-ISCCCLIBS =	../../lib/isccc/libisccc.la
-DNSLIBS =	../../lib/dns/libdns.la
+ISCLIBS =	../../lib/isc/libisc-export.la
+ISCCCLIBS =	../../lib/isccc/libisccc-export.la
+DNSLIBS =	../../lib/dns/libdns-export.la
 ISCCFGLIBS =	../../lib/cfg/libisccfg-export.la
 
-ISCDEPLIBS =	../../lib/isc/libisc.la
+ISCDEPLIBS =	../../lib/isc/libisc-export.la
 ISCCFGDEPLIBS =	libisccfg-export.la
 
 LIBS =		-ldl  -L../dns -L../isc -L../isccc
@@ -489,6 +489,7 @@
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} ${DNSLIBS} ${ISCCCLIBS} ${ISCLIBS}  \
 		${LIBS}
+	ln -sf .libs/libisccfg-export.so .
 
 timestamp: libisccfg-export.la
 	touch timestamp
diff -ur lib/samples/Makefile lib/samples/Makefile
--- lib/samples/Makefile	2016-03-16 19:44:04.832863999 +0000
+++ lib/samples/Makefile	2016-03-16 20:05:17.608863999 +0000
@@ -78,15 +78,15 @@
 CDEFINES =	-DVERSION=\"${VERSION}\" -DSYSCONFDIR=\"${sysconfdir}\"
 CWARNINGS =
 
-ISCLIBS =	../isc/libisc.la
-DNSLIBS =	../dns/libdns.la 
-ISCCFGLIBS =	../isccfg/libisccfg.la
-IRSLIBS =	../irs/libirs.la
+ISCLIBS =	../isc/libisc-export.la
+DNSLIBS =	../dns/libdns-export.la 
+ISCCFGLIBS =	../isccfg/libisccfg-export.la
+IRSLIBS =	../irs/libirs-export.la
 
-ISCDEPLIBS =	../isc/libisc.la
-DNSDEPLIBS =	../dns/libdns.la
-ISCCFGDEPLIBS =	../isccfg/libisccfg.la
-IRSDEPLIBS =	../irs/libirs.la
+ISCDEPLIBS =	../isc/libisc-export.la
+DNSDEPLIBS =	../dns/libdns-export.la
+ISCCFGDEPLIBS =	../isccfg/libisccfg-export.la
+IRSDEPLIBS =	../irs/libirs-export.la
 
 DEPLIBS =	${DNSDEPLIBS} ${ISCCFGDEPLIBS} ${ISCDEPLIBS}
 
diff -ur lib/tests/Makefile lib/tests/Makefile
--- lib/tests/Makefile	2016-03-16 19:44:04.756863999 +0000
+++ lib/tests/Makefile	2016-03-16 20:05:41.540863999 +0000
@@ -75,11 +75,11 @@
 CDEFINES =
 CWARNINGS =
 
-ISCLIBS =	../../lib/isc/libisc.la
-ISCCCLIBS =	../../lib/isccc/libisccc.la
+ISCLIBS =	../../lib/isc/libisc-export.la
+ISCCCLIBS =	../../lib/isccc/libisccc-export.la
 
-ISCDEPLIBS =	../../lib/isc/libisc.la
-ISCCCDEPLIBS =	libisccc.la
+ISCDEPLIBS =	../../lib/isc/libisc-export.la
+ISCCCDEPLIBS =	libisccc-export.la
 
 OBJS =		t_api.lo
 
